### PR TITLE
Support slots in cgx

### PR DESCRIPTION
**Pull request by berendkleinhaneveld**

### CODE REVIEW

File: collagraph/collagraph.py [hunk 1]
```diff
@@ -33,6 +36,15 @@ def create_element(type, props=None, *children) -> VNode:
     return VNode(type, reactive(props or {}), children or tuple(), key)
 
 
+def render_slot(name, props, slots):
```
Comment (berendkleinhaneveld) [inline collagraph/collagraph.py:39]: I figured that there should be `render_slot` function as well for use in functional components.

File: collagraph/collagraph.py [hunk 2]
```diff
@@ -206,12 +218,16 @@ def update_class_component(self, fiber: Fiber):
         self.reconcile_children(fiber, children)
 
     def update_function_component(self, fiber: Fiber):
-        children = [fiber.type(fiber.props)]
+        type_signature = inspect.signature(fiber.type)
```
Comment (berendkleinhaneveld) [inline collagraph/collagraph.py:221]: I'm not too sure about this one... Actually: I should just change this into a check whether the fiber.children is a dict. Should be faster and then you'll get an error if your render function doesn't support slots. Which is what you would expect.

**Comment by berendkleinhaneveld** [comment]
I should probably use a different name for 'default' content. Vue docs call it 'fallback' content which is more aptly named.

File: collagraph/collagraph.py (hunk 1 above)
Comment (Korijn) [reply collagraph/collagraph.py:38]: Do you have tests for this?

File: collagraph/collagraph.py [hunk 3]
```diff
@@ -19,6 +20,8 @@
 
 logger = logging.getLogger(__name__)
 
+VIRTUAL_NODES = {"template", "slot"}
```
Comment (Korijn) [inline collagraph/collagraph.py:22]: The name is a little confusing since you also have a VNode class

Comment (berendkleinhaneveld) [reply collagraph/collagraph.py:22]: Ah, right :) Would `VIRTUAL_NODE_TAGS` or `VIRTUAL_NODE_TYPES` work for you?

File: collagraph/collagraph.py (hunk 1 above)
Comment (berendkleinhaneveld) [reply collagraph/collagraph.py:38]: Yes, although not 100%. Now that I think about it: I could actually call this method from within `Component` in order to re-use the code. Might make more sense than adding tests.

File: collagraph/collagraph.py (hunk 3 above)
Comment (Korijn) [reply collagraph/collagraph.py:22]: Yeap!

Comment (berendkleinhaneveld) [reply collagraph/collagraph.py]: 👍 Changed it to `VIRTUAL_NODE_TYPES`, thanks!

**Pull request merged**